applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -372,253 +427,158 @@ void File::AddPath(const std::string& rPath)
+template<class TDataType>
+void File::WriteDataSet(
+    const std::string& rPath,
+    const TDataType& rData,
+    WriteInfo& rInfo)
 {
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
+    WriteDataSetImpl<TDataType, DataTransferMode::Collective>(rPath, rData, rInfo);
 }
 
-void File::WriteDataSetIndependent(const std::string& rPath, const Matrix<double>& rData, WriteInfo& rInfo)
+template<class TDataType>
+void File::WriteDataSetIndependent(
+    const std::string& rPath,
+    const TDataType& rData,
+    WriteInfo& rInfo)
 {
-    KRATOS_TRY;
-    KRATOS_ERROR << "Calling the base class method. Please override in the derived class." << std::endl;
-    KRATOS_CATCH("");
+    WriteDataSetImpl<TDataType, DataTransferMode::Independent>(rPath, rData, rInfo);
 }
 
 std::vector<unsigned> File::GetDataDimensions(const std::string& rPath) const
 {
     KRATOS_TRY;
-    constexpr int max_ndims = 5;
+
     int ndims;
-    hsize_t dims[max_ndims];
     hid_t dset_id, dspace_id;
     KRATOS_ERROR_IF((dset_id = H5Dopen(m_file_id, rPath.c_str(), H5P_DEFAULT)) < 0)
         << "H5Dopen failed." << std::endl;
     KRATOS_ERROR_IF((dspace_id = H5Dget_space(dset_id)) < 0)
         << "H5Dget_space failed." << std::endl;
     KRATOS_ERROR_IF((ndims = H5Sget_simple_extent_ndims(dspace_id)) < 0)
         << "H5Sget_simple_extent_ndims failed." << std::endl;
-    KRATOS_ERROR_IF(max_ndims < ndims) << "Maximum dimension: " << max_ndims
-                                       << ", dimension: " << ndims << std::endl;
-    KRATOS_ERROR_IF(H5Sget_simple_extent_dims(dspace_id, dims, nullptr) < 0)
+
+    std::vector<hsize_t> dims(ndims);
+    KRATOS_ERROR_IF(H5Sget_simple_extent_dims(dspace_id, dims.data(), nullptr) < 0)
         << "H5Sget_simple_extent_dims failed" << std::endl;
     KRATOS_ERROR_IF(H5Sclose(dspace_id) < 0) << "H5Sclose failed." << std::endl;
     KRATOS_ERROR_IF(H5Dclose(dset_id) < 0) << "H5Dclose failed." << std::endl;
 
-    return std::vector<unsigned>(dims, dims + ndims);
+    return std::vector<unsigned>(dims.begin(), dims.end());
+
     KRATOS_CATCH("");
 }
 
 bool File::HasIntDataType(const std::string& rPath) const
 {
-    KRATOS_TRY;
-    hid_t dset_id, dtype_id;
-    KRATOS_ERROR_IF((dset_id = H5Dopen(m_file_id, rPath.c_str(), H5P_DEFAULT)) < 0)
-        << "H5Dopen failed." << std::endl;
-    KRATOS_ERROR_IF((dtype_id = H5Dget_type(dset_id)) < 0)
-        << "H5Dget_type failed." << std::endl;
-    H5T_class_t type = H5Tget_class(dtype_id);
-    KRATOS_ERROR_IF(type == H5T_NO_CLASS) << "Invalid data type." << std::endl;
-    KRATOS_ERROR_IF(H5Tclose(dtype_id) < 0) << "H5Tclose failed." << std::endl;
-    KRATOS_ERROR_IF(H5Dclose(dset_id) < 0) << "H5Dclose failed." << std::endl;
-
-    return (type == H5T_INTEGER);
-    KRATOS_CATCH("");
+    return HasDataType<int>(rPath);
```

Comment:
ok so this is for backward compatibility. Nevermind my earlier comment then, sry for the noise.